File: apps/api/internal/dto/inventory.go
package dto

type InventoryRequest struct {
	Name         string  `json:"name" binding:"required"`
	Category     string  `json:"category"`
	Kind         string  `json:"kind" binding:"omitempty,oneof=ingredient stock"`
	Quantity     float64 `json:"quantity"`
	Unit         string  `json:"unit" binding:"required"`
	ReorderLevel float64 `json:"reorderLevel"`
	UnitCost     float64 `json:"unitCost" binding:"gte=0"`
}

File: packages/management/src/api/inventory.ts
import { secured } from './client';

export type InventoryItem = {
  id: number;
  name: string;
  category: string;
  kind: 'ingredient' | 'stock';
  quantity: number;
  unit: string;
  reorderLevel: number;
  unitCost: number;
  status: 'ready' | 'low' | 'out';
  imageUrl: string;
};

export const listInventory = (
  kind: 'ingredient' | 'stock',
  branchCode = 'SBC-AYA-001',
) =>
  secured<InventoryItem[]>(
    `/inventory?branchCode=${encodeURIComponent(branchCode)}&kind=${kind}`,
  );
